matlab/lib/+Ice/ConnectionInfo.m

```diff
@@ -2,43 +2,30 @@
     % ConnectionInfo   Base class providing access to the connection details.
     %
     % ConnectionInfo Properties:
-    %   underlying (Ice.ConnectionInfo) - The information of the underyling
-    %     transport or an empty array if there's no underlying transport.
-    %   incoming (logical) - Whether or not the connection is an incoming or
-    %     outgoing connection.
-    %   adapterName (char) - The name of the adapter associated with the
-    %     connection.
+    %   underlying (Ice.ConnectionInfo) - The information of the underlying transport or an empty array if there's no
+    %     underlying transport.
     %   connectionId (char) - The connection id.
 
     % Copyright (c) ZeroC, Inc. All rights reserved.
 
-    methods
-        function obj = ConnectionInfo(underlying, incoming, adapterName, connectionId)
-            if nargin == 0
-                underlying = [];
-                incoming = false;
-                adapterName = '';
-                connectionId = '';
-            end
-            obj.underlying = underlying;
-            obj.incoming = incoming;
-            obj.adapterName = adapterName;
-            obj.connectionId = connectionId;
-        end
-    end
-    properties(SetAccess=private)
-        % underlying   The information of the underyling transport or null
-        %   if there's no underlying transport.
+    properties(SetAccess=immutable)
+        % underlying   The information of the underlying transport or null if there's no underlying transport.
```

Comment:
For a separe PR:

You didn't update this comment but, seems that `null` isn't the correct term in MATLAB. The other comment above "or an empty array " seems more appropriate.  

Having two comments for the property seems unnecessary, the comment for the property should be placed with the properties section, see: https://www.mathworks.com/help/matlab/matlab_prog/create-help-for-classes.html